src/pylibsshext/session.pyx

```diff
@@ -383,6 +406,43 @@ cdef class Session(object):
         if rc == libssh.SSH_AUTH_ERROR or rc == libssh.SSH_AUTH_DENIED:
             raise LibsshSessionException("Failed to authenticate with password: %s" % self._get_session_error_str())
 
+    def authenticate_interactive(self, password):
+        """Authenticate this session using keyboard-interactive authentication.
+
+        :param password: The password to authenticate with.
+        :type password: str
+
+        :raises LibsshSessionException: If authentication failed.
+
+        :return: Nothing.
+        :rtype: NoneType
+        """
+        cdef int rc
+        cdef char should_echo
+        rc = libssh.ssh_userauth_kbdint(self._libssh_session, NULL, NULL)
+
+        while rc == libssh.SSH_AUTH_INFO:
```

Comment:
Does this need some retries limit/timeout?

Reply:
It isn't retries per se, and for a well behaving server I wouldn't think we would need to, but I'll add a check that makes sure we don't loop more than three times.

Reply:
Based on the libssh example here https://api.libssh.org/stable/libssh_tutor_authentication.html  for `The keyboard-interactive authentication method` loop if for the total number of prompts.

Reply:
There's also the outer `while (rc == SSH_AUTH_INFO)` loop. That page is largely what I was working from to understand what's going on so I could put this together.

Reply:
Ah..that's right. I missed the outer loop.